orchagent/intfsorch.cpp

```diff
@@ -550,6 +587,21 @@ void IntfsOrch::doTask(Consumer &consumer)
             }
             else
             {
+                /* Set nat zone id */
+                if (!nat_zone.empty())
+                {
+                    nat_zone_id = (uint32_t)stoul(nat_zone);
+                    if ((m_nat_zone.find(alias) == m_nat_zone.end()) or
+                        (nat_zone_id != m_nat_zone[alias]))
+                    {
+                        m_nat_zone[alias] = nat_zone_id;
+                        if (isNatSupported)
+                        {
+                            setRouterIntfsNatZoneId(port, nat_zone_id);
```

Comment:
I don't understand this call. The router interface is created only in setIntf. So this API is always going to return `true `after its first check.  You are actually setting the zone in as part of Line 808. Can you please check?

Reply:
User configures the new 'nat_zone' even after rif is created, this call is used to set the new 'nat_zone'.

Reply:
Got it but the set of RIF attribute is currently handled in setIntf.  It is done for SUB_PORT currently but I think can be modified to set NAT zone for any RIF type. Another follow up is, why do we need m_nat_zone map seperate? Can this nat zone be part of the port structure, like m_rif_id?

Reply:
Setting NAT zone for Physical, Vlan and Lag interfaces only, so not handled in setIntf.
Yes, nat zone can be part of port structure, made changes to address it.